### ISSUE TITLE
Expose Timeout exception

### ISSUE THREAD

**Issue by noschinl** [basvandijk/unbounded-delays#5]
I am using timeout together with QuickCheck. QuickCheck catches asynchronous exceptions and returns a "test failed" result then (including the interrupting exception). I would like to distinguish between failed tests and tests interrupted by a timeout. For this I need a way to tell whether an exception originated in the surrounding timeout. Hence I'd like to see a variant of timeout with signature

``` haskell
timeout' :: Integer -> (Timeout -> IO α) -> IO (Maybe α)
```

or (as one probably does not want to export the Timeout type)

``` haskell
timeout' :: Integer -> ((SomeException -> Bool) -> IO α) -> IO (Maybe α)
```

where `(SomeException -> Bool)` is a isTimeout function.

The use-case as the addition of soft timeouts in Tasty (see feuerbach/tasty#89) which still are recognized as timeouts by the system.

If this generalization is acceptable to you, I'd be willing to implement it.

**Comment by noschinl** [comment]
Hi @basvandijk -- any progress on that topic? I added an updated pull-request.

**Comment by feuerbach** [comment]
Hey Bas, could you implement this in some form?

**Comment by basvandijk** [comment]
This is now available in [unbounded-delays-0.1.1.0](http://hackage.haskell.org/package/unbounded-delays-0.1.1.0).

**Comment by feuerbach** [comment]
Thanks!
